src/compat_check.jl

```diff
@@ -151,10 +154,11 @@ function get_dep_entries end
             if haskey(getdeps(active_env.manifest), testreport_proj.deps[dep])
                 push!(deps_to_check, getdeps(active_env.manifest)[testreport_proj.deps[dep]])
             else
+                version_number = isa(testreport_proj.compat[dep], String) ? VersionNumber(testreport_proj.compat[dep]) : VersionNumber(testreport_proj.compat[dep].str)
```

Comment:
Can we do something like
```suggestion
                version_number = string(testreport_proj.compat[dep])
```

Reply:
In 1.7 `VersionNumber(testreport_proj.compat[dep]` is a `Pkg.Types.Compat` which is a new type that didn't exist, so `string(testreport_proj.compat[dep])` just produces something like `"Compat(VersionSpec(\"1.0\"), \"1.0\")"`. However I can make this line clearer

Reply:
Oh I see you mean not use `VersionNumber`. Yes quite possibly. I've been thinking about this and it feels like there must be a better solution/I'm not sure this is working exactly as intended. For this PR I'll keep using `VersionNumber` as it keeps the behaviour consistent, but I think a better way may be to to use `VersionSpec`.

Reply:
#79